sycl/source/device.cpp

```diff
@@ -29,7 +29,7 @@ void force_type(info::device_type &t, const info::device_type &ft) {
 }
 } // namespace detail
 
-device::device() : impl(detail::device_impl::getHostDeviceImpl()) {}
+device::device() { *this = default_selector{}.select_device(); }
```

Comment:
A some point the implementation should not use this deprecated feature from SYCL 1.2.1

Reply:
+1 on that, we should use `operator()` here. Also `*this = ` doesn't look good to me. I think, initializing impl with the impl of the selected device will be nicer.